app/models/rental.rb

```diff
@@ -26,6 +26,11 @@ class Rental < ActiveRecord::Base
   alias_attribute :start_date, :start_time
   alias_attribute :end_date, :end_time
 
+  scope :upcoming_rentals, -> { reserved.where('start_time <= ? AND end_time >= ?', DateTime.current + 1.day, DateTime.current) }
+  scope :all_future_rentals, -> { reserved.where('end_time >= ?', DateTime.current) }
+  scope :no_show_rentals, -> { reserved.where('end_time < ?', DateTime.current) }
+  scope :users_rentals, -> (current_user_id) { where(user_id: current_user_id) }
```

Comment:
Since you have

```ruby
belongs_to :user
```

in this model file, rather than a scope you could just write `some_user.rentals`.